src/plugins/copyCode.js

```diff
@@ -0,0 +1,77 @@
+const cheerio = module.parent.require('cheerio');
+
+const {
+  COPIED_TO_CLIPBOARD,
+  COPY_ICON,
+  COPY_TO_CLIPBOARD,
+} = require('../constants');
+
+function getButtonHTML() {
+  const html = `<button class="copy-btn-icon copy-btn" type="button">
+    <div class="copy-btn-body">
+    ${COPY_ICON}
+    <strong class="copy-btn-label">${COPY_TO_CLIPBOARD}</strong>
+    </div>
+    </button>`;
+  return html;
+}
+
+function getCopyCodeBlockScript() {
+  return `<script>
+    document.onclick = ((event) => {
+        let element = event.target;
+        const elementClassList = [...element.classList];
+        const listOfSelectors = ['copy-btn-label', 'copy-btn', 'copy-btn-icon', 'copy-btn-body', 'cpy'];
+        let parents = [];
+        let copyButton = null;
+        while (element) {
+            parents.unshift(element);
+            let classList = element.classList;
+            if (classList) {
+                classList.forEach((value, key, obj) => {
+                    if (value === 'copy-btn') {
+                        copyButton = element;
+                    }
+                });
+            }
+            element = element.parentNode;
+        }
+        
+        const copyButtonClicked = listOfSelectors.some(sel => elementClassList.includes(sel));
+        if (copyButtonClicked) {
+            var pre;
+            for (const obj of parents) {
+                    if (obj['outerHTML'] && obj['outerHTML'].indexOf('<pre>') === 0) {
+                    pre = obj;
+                }
+            }
+            let codeElement = pre.querySelector('code');
+            const textElement = document.createElement('textarea');
+            textElement.value = codeElement.textContent;
```

Comment:
I wonder if we can just select the codeElement directly and do the copy without having to create the dummy text area. If that is not possible, let's add some comments here to explain why we are creating the dummy text area.

Reply:
The reason I used the dummy text area is because `select()` is defined for `<textarea>` and `<input>` elements. I can add comments to explain the usage 👍